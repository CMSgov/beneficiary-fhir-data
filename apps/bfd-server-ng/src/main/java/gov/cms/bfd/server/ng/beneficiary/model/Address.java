package gov.cms.bfd.server.ng.beneficiary.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import java.util.Optional;

@Embeddable
public class Address {
  @Column(name = "geo_usps_state_cd")
  private String stateCode;

  @Column(name = "geo_zip5_cd")
  private String zipCode;

  @Column(name = "geo_zip_plc_name")
  private String city;

  @Column(name = "bene_line_1_adr")
  private Optional<String> addressLine1;

  @Column(name = "bene_line_2_adr")
  private Optional<String> addressLine2;

  @Column(name = "bene_line_3_adr")
  private Optional<String> addressLine3;

  @Column(name = "bene_line_4_adr")
  private Optional<String> addressLine4;

  @Column(name = "bene_line_5_adr")
  private Optional<String> addressLine5;

  @Column(name = "bene_line_6_adr")
  private Optional<String> addressLine6;

  org.hl7.fhir.r4.model.Address toFhir() {
    var address =
        new org.hl7.fhir.r4.model.Address()
            .setState(stateCode)
            .setPostalCode(zipCode)
            .setCity(city);
    addressLine1.ifPresent(address::addLine);
    addressLine2.ifPresent(address::addLine);
    addressLine3.ifPresent(address::addLine);
    addressLine4.ifPresent(address::addLine);
    addressLine5.ifPresent(address::addLine);
    addressLine6.ifPresent(address::addLine);

    return address;
  }
}
