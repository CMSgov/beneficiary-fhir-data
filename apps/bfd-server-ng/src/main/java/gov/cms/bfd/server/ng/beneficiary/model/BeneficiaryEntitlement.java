package gov.cms.bfd.server.ng.beneficiary.model;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.ZonedDateTime;
import lombok.Getter;

/** Entity representing BeneficiaryEntitlement. */
@Entity
@Getter
@Table(name = "beneficiary_entitlement", schema = "idr")
public class BeneficiaryEntitlement {

  @Id
  @Column(name = "bene_sk", nullable = false)
  private Long beneSk;

  @Column(name = "bene_rng_bgn_dt", nullable = false)
  private LocalDate benefitRangeBeginDate;

  @Column(name = "bene_rng_end_dt", nullable = false)
  private LocalDate benefitRangeEndDate;

  @Column(name = "bene_mdcr_entlmt_type_cd", length = 1)
  private String medicareEntitlementTypeCode;

  @Column(name = "bene_mdcr_enrlmt_rsn_cd", length = 1)
  private String medicareEnrollmentReasonCode;

  @Column(name = "bene_mdcr_entlmt_stus_cd", length = 1)
  private String medicareEntitlementStatusCode;

  @Column(name = "idr_trans_efctv_ts", nullable = false)
  private ZonedDateTime idrTransEffectiveTimestamp;

  @Column(name = "idr_trans_obslt_ts", nullable = false)
  private ZonedDateTime idrTransObsoleteTimestamp;

  @Column(name = "idr_updt_ts", nullable = false)
  private ZonedDateTime idrUpdateTimestamp;

  @Column(name = "bfd_created_ts", nullable = false)
  private ZonedDateTime bfdCreatedTimestamp;

  @Column(name = "bfd_updated_ts", nullable = false)
  private ZonedDateTime bfdUpdatedTimestamp;
}
