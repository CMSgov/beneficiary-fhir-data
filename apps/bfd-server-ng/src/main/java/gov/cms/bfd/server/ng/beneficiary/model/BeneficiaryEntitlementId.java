package gov.cms.bfd.server.ng.beneficiary.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import java.io.Serializable;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

/** Represents the composite primary key for the {@link BeneficiaryEntitlement} entity. */
@EqualsAndHashCode
@NoArgsConstructor
@Getter
@AllArgsConstructor
@Embeddable
public class BeneficiaryEntitlementId implements Serializable {

  @Column(name = "bene_sk")
  private Long beneSk;

  @Column(name = "bene_rng_bgn_dt")
  private LocalDate benefitRangeBeginDate;

  @Column(name = "bene_rng_end_dt")
  private LocalDate benefitRangeEndDate;

  @Column(name = "bene_mdcr_entlmt_type_cd")
  private String medicareEntitlementTypeCode;
}
