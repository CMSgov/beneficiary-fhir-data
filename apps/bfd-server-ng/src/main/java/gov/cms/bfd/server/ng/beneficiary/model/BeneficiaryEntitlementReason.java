package gov.cms.bfd.server.ng.beneficiary.model;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.ZonedDateTime;
import lombok.Getter;

/** Entity representing BeneficiaryEntitlementReason. */
@Entity
@Getter
@Table(name = "beneficiary_entitlement_reason", schema = "idr")
public class BeneficiaryEntitlementReason {

  @Id
  @Column(name = "bene_sk", nullable = false)
  private Long beneSk;

  @Column(name = "bene_rng_bgn_dt", nullable = false)
  private LocalDate benefitRangeBeginDate;

  @Column(name = "bene_rng_end_dt", nullable = false)
  private LocalDate benefitRangeEndDate;

  @Column(name = "idr_trans_efctv_ts", nullable = false)
  private ZonedDateTime idrTransEffectiveTimestamp;

  @Column(name = "bene_mdcr_entlmt_rsn_cd", length = 1)
  private String medicareEntitlementReasonCode;

  @Column(name = "idr_trans_obslt_ts", nullable = false)
  private ZonedDateTime idrTransObsoleteTimestamp;

  @Column(name = "idr_updt_ts")
  private ZonedDateTime idrUpdateTimestamp;

  @Column(name = "bfd_created_ts", nullable = false)
  private ZonedDateTime bfdCreatedTimestamp;

  @Column(name = "bfd_updated_ts", nullable = false)
  private ZonedDateTime bfdUpdatedTimestamp;
}
