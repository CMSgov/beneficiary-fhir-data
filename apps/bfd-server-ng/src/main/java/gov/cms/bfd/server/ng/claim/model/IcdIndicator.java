package gov.cms.bfd.server.ng.claim.model;

import gov.cms.bfd.server.ng.SystemUrls;
import java.util.Arrays;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum IcdIndicator {
  DEFAULT("", SystemUrls.CMS_ICD_9, SystemUrls.ICD_9_CM),
  ICD_9("9", SystemUrls.CMS_ICD_9, SystemUrls.ICD_9_CM),
  ICD_10("0", SystemUrls.CMS_ICD_10, SystemUrls.ICD_10_CM);

  public static IcdIndicator fromCode(String code) {
    return Arrays.stream(values()).filter(v -> v.code.equals(code)).findFirst().get();
  }

  private final String code;
  private final String procedureSystem;
  private final String diagnosisSytem;
}
