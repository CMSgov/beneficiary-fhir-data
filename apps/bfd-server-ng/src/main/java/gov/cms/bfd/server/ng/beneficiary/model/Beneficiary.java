package gov.cms.bfd.server.ng.beneficiary.model;

import gov.cms.bfd.server.ng.DateUtil;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.util.List;
import java.util.Optional;
import org.hl7.fhir.r4.model.Patient;

/** Beneficiary table representation. */
@Entity
@Table(name = "beneficiary", schema = "idr")
public class Beneficiary {
  @Id
  @Column(name = "bene_sk", nullable = false)
  private long beneSk;

  @Column(name = "bene_xref_efctv_sk_computed", nullable = false)
  private long xrefSk;

  @Column(name = "bene_mbi_id", nullable = false)
  private String mbi;

  @Column(name = "bene_brth_dt", nullable = false)
  private LocalDate birthDate;

  @Column(name = "bene_race_cd", nullable = false)
  private RaceCode raceCode;

  @Column(name = "bene_sex_cd")
  private Optional<SexCode> sexCode;

  @Column(name = "cntct_lang_cd", nullable = false)
  private LanguageCode languageCode;

  @Embedded private Name beneficiaryName;
  @Embedded private Address address;
  @Embedded private Meta meta;
  @Embedded private DeathDate deathDate;

  /**
   * Transforms the beneficiary record to its FHIR representation.
   *
   * @return patient record
   */
  public Patient toFhir() {
    var patient = new Patient();
    patient.setId(String.valueOf(beneSk));

    patient.setName(List.of(beneficiaryName.toFhir()));
    patient.setBirthDate(DateUtil.toDate(birthDate));
    address.toFhir().ifPresent(a -> patient.setAddress(List.of(a)));
    sexCode.ifPresent(
        s -> {
          patient.setGender(s.toFhirAdministrativeGender());
          patient.addExtension(s.toFhirSexExtension());
        });

    patient.setCommunication(List.of(languageCode.toFhir()));
    deathDate.toFhir().ifPresent(patient::setDeceased);
    patient.addExtension(raceCode.toFhir());
    patient.setMeta(meta.toFhir());

    return patient;
  }
}
