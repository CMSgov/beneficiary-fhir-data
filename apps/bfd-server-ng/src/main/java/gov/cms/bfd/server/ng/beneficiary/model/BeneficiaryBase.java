package gov.cms.bfd.server.ng.beneficiary.model;

import gov.cms.bfd.server.ng.util.DateUtil;
import gov.cms.bfd.server.ng.util.SystemUrls;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.Transient;
import java.time.LocalDate;
import java.time.ZonedDateTime;
import java.util.List;
import java.util.Optional;
import java.util.UUID;
import lombok.Getter;
import lombok.NonNull;
import org.hl7.fhir.r4.model.Patient;

/**
 * Base class for beneficiary entities. This is used to prevent excessive fetching for patient
 * information.
 */
@Getter
@MappedSuperclass
public abstract class BeneficiaryBase {
  @Id
  @Column(name = "bene_sk")
  protected long beneSk;

  @Column(name = "bene_xref_efctv_sk_computed")
  protected long xrefSk;

  @Column(name = "bene_brth_dt")
  protected LocalDate birthDate;

  @Column(name = "bene_race_cd")
  protected RaceCode raceCode;

  @Column(name = "bene_sex_cd")
  protected Optional<SexCode> sexCode;

  @Column(name = "cntct_lang_cd")
  protected LanguageCode languageCode;

  @Column(name = "idr_trans_obslt_ts")
  protected ZonedDateTime obsoleteTimestamp;

  @Embedded protected Name beneficiaryName;
  @Embedded protected Address address;
  @Embedded protected Meta meta;
  @Embedded protected DeathDate deathDate;
  @Embedded protected CurrentIdentifier identifier;
  @Transient protected String id = UUID.randomUUID().toString();

  /**
   * Determines if this beneficiary has been merged into another.
   *
   * @return whether the beneficiary is merged
   */
  public boolean isMergedBeneficiary() {
    return beneSk != xrefSk;
  }

  /**
   * Convenience method to convert to FHIR Patient with a specific profile.
   *
   * @param profile the FHIR profile to apply
   * @return patient record
   */
  public Patient toFhir(String profile) {
    var patient = new Patient();

    if (profile.equals(SystemUrls.PROFILE_C4DIC_PATIENT)) {
      patient.setId(id);
    } else {
      patient.setId(String.valueOf(beneSk));
    }

    // Only return a skeleton resource or merged beneficiaries
    if (isMergedBeneficiary()) {
      return patient;
    }

    patient.setCommunication(List.of(languageCode.toFhir()));
    deathDate.toFhir().ifPresent(patient::setDeceased);
    patient.setName(List.of(beneficiaryName.toFhir()));
    patient.setBirthDate(DateUtil.toDate(birthDate));
    address.toFhir().ifPresent(a -> patient.setAddress(List.of(a)));
    sexCode.ifPresent(
            s -> {
              patient.setGender(s.toFhirAdministrativeGender());
              patient.addExtension(s.toFhirSexExtension());
            });
    patient.addExtension(raceCode.toFhir());
    patient.setMeta(meta.toFhirPatient(profile));

    return patient;
  }
}
