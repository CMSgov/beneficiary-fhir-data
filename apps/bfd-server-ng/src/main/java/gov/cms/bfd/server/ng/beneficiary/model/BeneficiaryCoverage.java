package gov.cms.bfd.server.ng.beneficiary.model;

import gov.cms.bfd.server.ng.input.CoverageCompositeId;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import java.util.Collection;
import java.util.Optional;
import java.util.Set;
import java.util.stream.Stream;
import lombok.Getter;
import org.hl7.fhir.r4.model.Coverage;
import org.hl7.fhir.r4.model.Organization;
import org.hl7.fhir.r4.model.Reference;

/** Entity representing the beneficiary table with coverage info. */
@Entity
@Getter
@Table(name = "beneficiary", schema = "idr")
public class BeneficiaryCoverage extends BeneficiaryBase {
  @OneToMany(fetch = FetchType.EAGER)
  @JoinColumn(name = "bene_sk")
  private Set<BeneficiaryEntitlement> beneficiaryEntitlements;

  @OneToMany(fetch = FetchType.EAGER)
  @JoinColumn(name = "bene_sk")
  private Set<BeneficiaryThirdParty> beneficiaryThirdParties;

  @OneToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "bene_sk")
  private BeneficiaryStatus beneficiaryStatus;

  @OneToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "bene_sk")
  private BeneficiaryEntitlementReason beneficiaryEntitlementReason;

  private Optional<BeneficiaryEntitlementReason> getEntitlementReason() {
    return Optional.ofNullable(beneficiaryEntitlementReason);
  }

  private Optional<BeneficiaryStatus> getStatus() {
    return Optional.ofNullable(beneficiaryStatus);
  }

  /**
   * Creates a FHIR Coverage resource.
   *
   * @param coverageCompositeId The full ID for the Coverage resource.
   * @return A FHIR Coverage object.
   */
  public Coverage toFhir(CoverageCompositeId coverageCompositeId) {
    var coverage = new Coverage();
    coverage.setId(coverageCompositeId.fullId());

    var coveragePart = coverageCompositeId.coveragePart();
    var coverageType = coveragePart.getStandardCode();
    var entitlementOpt =
        beneficiaryEntitlements.stream()
            .filter(
                e -> e.getId().getMedicareEntitlementTypeCode().toUpperCase().equals(coverageType))
            .findFirst();
    // If no entitlement record, bene doesn't have this coverage type
    if (entitlementOpt.isEmpty()) {
      return coverage;
    }
    var entitlement = entitlementOpt.get();

    coverage.setMeta(meta.toFhirCoverage());

    coverage.setBeneficiary(new Reference("Patient/" + beneSk));

    coverage.setRelationship(RelationshipFactory.createSelfSubscriberRelationship());

    Organization cmsOrg = OrganizationFactory.createCmsOrganization();
    coverage.addContained(cmsOrg);

    coverage.addPayor(new Reference().setReference("#" + cmsOrg.getIdElement().getIdPart()));

    identity.toFhirMbiIdentifier().ifPresent(coverage::addIdentifier);
    coverage.setSubscriberId(identity.getMbiValue());

    coverage.setType(coveragePart.toFhirTypeCode());
    coverage.addClass_(coveragePart.toFhirClassComponent());
    var fhirPeriodOpt = entitlement.toFhirPeriod();
    fhirPeriodOpt.ifPresent(coverage::setPeriod);
    coverage.setStatus(Coverage.CoverageStatus.ACTIVE);

    var beneficiaryThirdParty =
        beneficiaryThirdParties.stream()
            .filter(tp -> tp.getId().getThirdPartyTypeCode().toUpperCase().equals(coverageType))
            .findFirst();

    Stream.of(
            beneficiaryThirdParty.map(BeneficiaryThirdParty::toFhirExtensions),
            Optional.of(entitlement.toFhirExtensions()),
            getStatus().map(BeneficiaryStatus::toFhirExtensions),
            getEntitlementReason().map(BeneficiaryEntitlementReason::toFhirExtensions))
        .flatMap(Optional::stream)
        .flatMap(Collection::stream)
        .forEach(coverage::addExtension);

    return coverage;
  }

  /**
   * Creates a FHIR Coverage resource or None if the beneficiary does not have the matching coverage
   * type.
   *
   * @param coverageCompositeId The full ID for the Coverage resource.
   * @return A FHIR Coverage object.
   */
  public Optional<Coverage> toFhirCoverageIfPresent(CoverageCompositeId coverageCompositeId) {
    var coverage = toFhir(coverageCompositeId);
    if (coverage.getIdentifier().isEmpty()) {
      return Optional.empty();
    }
    return Optional.of(coverage);
  }
}
