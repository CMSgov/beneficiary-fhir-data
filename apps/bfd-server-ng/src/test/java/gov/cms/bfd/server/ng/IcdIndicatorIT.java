package gov.cms.bfd.server.ng;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;

import ca.uhn.fhir.rest.gclient.IReadTyped;
import gov.cms.bfd.server.ng.claim.model.IcdIndicator;
import java.util.List;
import java.util.stream.Collectors;
import org.hl7.fhir.r4.model.ExplanationOfBenefit;
import org.junit.jupiter.api.Test;

// Integration test for ICD code formatting.
public class IcdIndicatorIT extends IntegrationTestBase {
  private static final String FORMATTED_ICD_9_CODE_25000 = "250.00";
  private static final String FORMATTED_ICD_9_CODE_E8889 = "E888.9";
  private static final String FORMATTED_ICD_9_CODE_V1005 = "V10.05";
  private static final String RAW_30495 = "30495";
  private static final String RAW_25000 = "25000";
  private static final String RAW_E8889 = "E8889";
  private static final String RAW_V1005 = "V1005";
  private static final String FORMATTED_ICD_9_CODE_30495 = "304.95";

  private IReadTyped<ExplanationOfBenefit> eobRead() {
    return getFhirClient().read().resource(ExplanationOfBenefit.class);
  }

  @Test
  void eobReadAndIcd9Formatting() {
    var eob = eobRead().withId(CLAIM_ID_ADJUDICATED_ICD_9).execute();
    assertFalse(eob.isEmpty());

    // Extract formatted diagnosis codes from the EOB.
    var eobCodes =
        eob.getDiagnosis().stream()
            .map(d -> d.getDiagnosisCodeableConcept().getCodingFirstRep().getCode())
            .toList();

    var rawCodes = List.of(RAW_E8889, RAW_V1005, RAW_30495, RAW_25000);
    var independentlyFormatted =
        rawCodes.stream()
            .collect(
                Collectors.toMap(
                    c -> c,
                    IcdIndicator.ICD_9::formatDiagnosisCode,
                    (a, b) -> a,
                    java.util.LinkedHashMap::new));

    var expectedOrder =
        List.of(
            FORMATTED_ICD_9_CODE_30495,
            FORMATTED_ICD_9_CODE_25000,
            FORMATTED_ICD_9_CODE_E8889,
            FORMATTED_ICD_9_CODE_V1005);
    assertEquals(expectedOrder, eobCodes, "ICD-9 codes should be formatted and ordered");
    assertEquals(FORMATTED_ICD_9_CODE_25000, independentlyFormatted.get(RAW_25000));
    assertEquals(FORMATTED_ICD_9_CODE_E8889, independentlyFormatted.get(RAW_E8889));
    assertEquals(FORMATTED_ICD_9_CODE_V1005, independentlyFormatted.get(RAW_V1005));

    // assert string representation instead of a snapshot.
    var expectedMap = new java.util.LinkedHashMap<String, String>();
    expectedMap.put(RAW_E8889, FORMATTED_ICD_9_CODE_E8889);
    expectedMap.put(RAW_V1005, FORMATTED_ICD_9_CODE_V1005);
    expectedMap.put(RAW_30495, FORMATTED_ICD_9_CODE_30495);
    expectedMap.put(RAW_25000, FORMATTED_ICD_9_CODE_25000);

    var actualSnapshotString = eobCodes.toString() + "|" + independentlyFormatted.toString();
    var expectedSnapshotString = expectedOrder.toString() + "|" + expectedMap.toString();
    assertEquals(
        expectedSnapshotString,
        actualSnapshotString,
        "Compact string comparison of codes and formatted map should match");
  }

  @Test
  void eobContainsAllFormattedIcd9DiagnosisCodes() {
    var eob = eobRead().withId(CLAIM_ID_ADJUDICATED_ICD_9).execute();
    var codes =
        eob.getDiagnosis().stream()
            .map(d -> d.getDiagnosisCodeableConcept().getCodingFirstRep().getCode())
            .toList();

    var expected =
        List.of(
            FORMATTED_ICD_9_CODE_30495,
            FORMATTED_ICD_9_CODE_25000,
            FORMATTED_ICD_9_CODE_E8889,
            FORMATTED_ICD_9_CODE_V1005);
    assertEquals(expected, codes, "ICD-9 diagnosis codes should be formatted and ordered");

    // compare string representations rather than using snapshot storage.
    assertEquals(expected.toString(), codes.toString(), "Compact codes string should match");
  }
}
