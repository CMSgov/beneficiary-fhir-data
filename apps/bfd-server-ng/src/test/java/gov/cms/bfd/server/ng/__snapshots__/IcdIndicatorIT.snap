gov.cms.bfd.server.ng.IcdIndicatorIT.eobContainsAllFormattedIcd9DiagnosisCodes[icd9DiagnosisCodes]=[
{codes=[304.92, 250.00, E888.9, V10.05]}
]


gov.cms.bfd.server.ng.IcdIndicatorIT.eobReadAndIcd9Formatting[eobAndIcd9Formats]=[
{eobDiagnosisCodes=[304.92, 250.00, E888.9, V10.05], formattedFromIndicator={30492=304.92, 25000=250.00, E8889=E888.9, V1005=V10.05}}
]


gov.cms.bfd.server.ng.IcdIndicatorIT.snapshotFormattedCodes[icdFormattedValues]=[
{icd9Proc=12.345, icd10=F10.10, icd9Diag=123.45}
]